File: backend/routers/conquistas.py
"""
Router de Conquistas — as medalhas da tela 34.

Endpoints:
  GET /conquistas   estado das conquistas do usuário

Só leitura, e de propósito: as conquistas são derivadas do histórico a cada
chamada (ver services/achievements.py), então não há o que criar nem
atualizar. O copy de cada uma (nome, frase, ícone) mora no frontend, junto do
design; aqui só sai a verdade — se foi ganha, quando, e onde a pessoa está.
"""

from fastapi import APIRouter

from models import AchievementList
from services.achievements import conquistas_do_usuario
from services.config import settings

router = APIRouter(prefix="/conquistas", tags=["conquistas"])


@router.get("", response_model=AchievementList)
def listar_conquistas():
    items = conquistas_do_usuario(settings.MVP_USER_ID)
    return {"items": items, "ganhas": sum(1 for i in items if i["ganha"]), "total": len(items)}

File: supabase/functions/api/routes/conquistas.ts
/*
  Rota de Conquistas — porta de routers/conquistas.py.

  GET /conquistas   estado das conquistas do usuário

  Só leitura: as conquistas são derivadas do histórico a cada chamada (ver
  _shared/achievements.ts), não há o que criar nem atualizar.
*/

import type { Router } from "../../_shared/router.ts";
import { currentUserId } from "../../_shared/auth.ts";
import { conquistasDoUsuario } from "../../_shared/achievements.ts";

export function register(router: Router): void {
  router.get("/conquistas", async (ctx) => {
    const userId = await currentUserId(ctx.req);
    const items = await conquistasDoUsuario(userId);
    return { items, ganhas: items.filter((i) => i.ganha).length, total: items.length };
  });
}
